### ISSUE TITLE
Thread CppFileParserTh leads to SegFault

### ISSUE THREAD

**Issue by Wilhelm-Lee** [royqh1979/RedPanda-CPP#155]
## Replay
When moving cursor (I-beam editing cursor) towards a line highlighted with clang shows the including file does not exist after modifing the file to making it `unsaved`, the whole programme exited in a sudden.

[Image](https://ibb.co/9gcppv2)

## Handling
After debugging with gdb, the following output is noticed when SegFault is triggered.
```
Thread 46 "CppFileParserTh" received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7f019223f6c0 (LWP 2050422)]
0x00005570a3560df1 in QMap<QString, bool>::keys() const ()
```

Also, dragging some certain files from sidebar into field of editor, or, opening a file that does not exist, yet, registered in `.dev` file, could cause the same error to occur as I metioned above.  
Something alike as following.
```
Thread 19 "CppFileParserTh" received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7f2cc2dfd6c0 (LWP 2051502)]
0x0000555cde31edf1 in QMap<QString, bool>::keys() const ()
```

Clearly, the problem is remarkably similar to my last updated issue. Good thing that we could see it today.

Video recorded is uploaded at:
[LINK](https://www.bilibili.com/video/BV1Kc411o7Q4)

**Comment by royqh1979** [comment]
Thanks for the bug report.

It will be fixed in the next commit.

William Lee ***@***.***> 于2023年11月4日周六 06:41写道：

**Comment by royqh1979** [comment]
What's the version of redpand C++ that you are using?

And can you share your project with me?

It seems like a null pointer visit. I've tried to fix it, but I can't
reproduce it, so I can't affirm that the fix is worked.

William Lee ***@***.***> 于2023年11月4日周六 06:41写道：

**Comment by Wilhelm-Lee** [comment]
Sorry for giving you the reply so late.

Unfortunately, I have gone for a trip to Beijing, and redpanda is currently at home, which are not available to commit here and now.

The project that causes the issue is now, however, modified.  And the problem that make clang complain is now long gone.  But I think I can give you the latest version of my project, not mentioning it open source, of course.

So, I will git the project up at sometime during today.

And there is another thing.  My laptop does have a redpanda, though. But I cannot confirm if the version matches the one I used at home. For now, it's `2.26-alpha6`, working fine on my laptop, despite I didn't test it for similar cases.  I got this panda by git clone the latest version yesterday, by the way.

I will come home on 11th Nov, 2023, and I will contact you again.
